### PR TITLE
Update STATE.md

### PR DESCRIPTION
**Pull request by seadowg**
I wanted to give this a spruce up as we get towards the end of the release cycle. Interested to know if others spot things we can change, delete or add!

### CODE REVIEW

File: docs/STATE.md [hunk 1]
```diff
@@ -21,27 +19,29 @@ and update this document as the code evolves.
...
+* Writing all new code in Kotlin
 * Writing new code using a [multi-module approach](CODE-GUIDELINES.md#gradle-sub-modules) (feature modules, mini frameworks etc) and breaking old code out into modules when opportunities come up
 * Trying to remove technical debt flagged with `@Deprecated`
-* Replacing async work such as `AsyncTask` with `LiveData` + `Scheduler` abstraction
+* Replacing async work such as `AsyncTask` with `Flow`/`LiveData` + `Scheduler` abstraction
 * Gradually removing use of `CursorLoader` (all remaining uses are in `CursorLoaderFactory`)
 * Using AndroidX Test in new local tests and migrating other local tests as we touch them (from classic Robolectric)
-* Moving towards a ["data services"](data_services_architecture.pdf) oriented architecture that has emerged over time
+* Moving towards a ["data services"](data_services_architecture.pdf) oriented architecture that has emerged over time that uses AndroidX Architecture Components for the core of the UI (Fragment, View, ViewModel etc)
+* Improving the `MapFragment` abstraction so more logic can be shared between the map engines
```
Comment (seadowg) [inline docs/STATE.md:47]: We're not actually doing this yet, but I really want to start.

**Review by lognaturel — APPROVED** [review]

File: docs/STATE.md [hunk 2]
```diff
@@ -21,27 +19,29 @@ and update this document as the code evolves.
...
+* Moving responsibilities out of `FormFillingActivity`
+* Writing all new code in Kotlin
 * Writing new code using a [multi-module approach](CODE-GUIDELINES.md#gradle-sub-modules) (feature modules, mini frameworks etc) and breaking old code out into modules when opportunities come up
 * Trying to remove technical debt flagged with `@Deprecated`
-* Replacing async work such as `AsyncTask` with `LiveData` + `Scheduler` abstraction
+* Replacing async work such as `AsyncTask` with `Flow`/`LiveData` + `Scheduler` abstraction
 * Gradually removing use of `CursorLoader` (all remaining uses are in `CursorLoaderFactory`)
 * Using AndroidX Test in new local tests and migrating other local tests as we touch them (from classic Robolectric)
-* Moving towards a ["data services"](data_services_architecture.pdf) oriented architecture that has emerged over time
+* Moving towards a ["data services"](data_services_architecture.pdf) oriented architecture that has emerged over time that uses AndroidX Architecture Components for the core of the UI (Fragment, View, ViewModel etc)
```
Comment (grzesiek2010) [inline docs/STATE.md]: `etc` should have a dot at the end: `etc.`

File: docs/STATE.md [hunk 3]
```diff
@@ -21,27 +19,29 @@ and update this document as the code evolves.
 * UI is "iconic" (old) but with a lot of inconsistencies and quirks and is best adapted to small screens (although often used on tablets)
 * A lot of code lives in between one "god" Activity (`FormFillingActivity`) and a process singleton (`FormController`)
 * Core form entry flow uses custom side-to-side swipe view (in `FormFillingActivity` made up of `ODKView`)
-* Questions are rendered using a view "framework" of implementations inheriting from `QuestionWidget` (which is documented at in [WIDGETS.MD](WIDGETS.md))
+* Questions are rendered using a view "framework" of implementations inheriting from `QuestionWidget` (which is documented at in [WIDGETS.MD](WIDGETS.md)) which are also used to store UI state during form entry
```
Comment (grzesiek2010) [inline docs/STATE.md]: Is `at in` correct in English? It's not my fist language but as far as I know, it should be just in.

File: docs/STATE.md [hunk 4]
```diff
@@ -21,27 +19,29 @@ and update this document as the code evolves.
...
 * Dagger2 is used to inject "black box" objects such as Activity and just uses a very basic setup
 * Http is handled using OkHttp3 and https client abstractions are generally wrapped in Android's AsyncTask
-* Geo activities use three engines (Mapbox, osmdroid, Google Maps) depending on the selected basemap even though Mapbox could do everything osmdroid does
+* Geo activities use three engines (Mapbox, osmdroid, Google Maps) depending on the selected basemap
 * Code goes through static analysis using CheckStyle, PMD, ktlint and Android Lint
 * Forms get into the app from two different sources (Open Rosa servers and disk) but the logic for this is disparate and they don't sit behind a common interface
 * Instances are linked to the forms they are instances of through formid and version. However, the same formid and version combination could represent multiple forms in storage
 * `SharedPreferences` is wrapped in app's own `Settings` abstraction
+* The form hierarchy is rendered using `FormHierarchyActivity` which hasn't been seriously touched (at a code or design) level for a few years
```
Comment (grzesiek2010) [inline docs/STATE.md]: There should be a comma before which: `...using FormHierarchyActivity, which...`

**Review by grzesiek2010 — CHANGES_REQUESTED** [review]

File: docs/STATE.md (hunk 2 above)
Comment (seadowg) [reply docs/STATE.md]: I didn't know that!

File: docs/STATE.md (hunk 3 above)
Comment (seadowg) [reply docs/STATE.md]: No, that's a typo. It could be either "document at" or "documented in" - my brain seems to have dumped out both.

**Review by grzesiek2010 — APPROVED** [review]

**Pull request merged**